Sources/Private/CoreAnimation/Animations/LayerProperty.swift

```diff
@@ -159,6 +159,34 @@ extension LayerProperty {
       },
       customizableProperty: nil /* currently unsupported */ )
   }
+
+  static var shadowOpacity: LayerProperty<CGFloat> {
+    .init(
+      caLayerKeypath: #keyPath(CALayer.shadowOpacity),
+      defaultValue: 0,
+      customizableProperty: nil /* currently unsupported */ )
+  }
+
+  static var shadowColor: LayerProperty<CGColor> {
+    .init(
+      caLayerKeypath: #keyPath(CALayer.shadowColor),
+      defaultValue: .rgb(0, 0, 0),
+      customizableProperty: nil /* currently unsupported */ )
+  }
+
+  static var shadowRadius: LayerProperty<CGFloat> {
+    .init(
+      caLayerKeypath: #keyPath(CALayer.shadowRadius),
+      defaultValue: 3.0,
+      customizableProperty: nil /* currently unsupported */ )
+  }
+
+  static var shadowOffset: LayerProperty<CGSize> {
+    .init(
+      caLayerKeypath: #keyPath(CALayer.shadowOffset),
+      defaultValue: CGSize(width: 0, height: -3.0),
```

Comment:
why does this and `shadowRadius` have non-zero default values?

Reply:
This is the default value of the `CALayer.shadowOffset` property ([documentation](https://developer.apple.com/documentation/quartzcore/calayer/1410970-shadowoffset):
> The default value of this property is `(0.0, -3.0)`.

 Knowing this value lets us apply certain optimizations in the code that generates the `CAKeyframe`s. (e.g. if the Lottie animation just sets the property to the default value, we can completely omit the keyframes and reduce the overhead of computing the keyframes)

Reply:
ah great, this might be a good comment to add in code here so others aren't confused in the future!